pkgs/development/python-modules/wxPython/4.0.nix

```diff
@@ -1,35 +1,83 @@
 { lib
+, stdenv
+, openglSupport ? true
+, libX11
+, pyopengl
 , buildPythonPackage
 , fetchPypi
 , pkgconfig
 , gtk3
 , libjpeg
 , libtiff
+, wxmac
+, darwin
 , SDL
 , gst-plugins-base
 , libnotify
 , freeglut
 , xorg
 , which
+, cairo
+, pango
+, python3
+, doxygen
+, ncurses
+, libpng
+, gstreamer
+, wxGTK30
+, python3Packages
 }:
 
+let
+wxGTK = wxGTK30.override { withGtk2 = false; withWebKit = true; };
+in
 buildPythonPackage rec {
   pname = "wxPython";
   version = "4.0.4";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0d9ef4260cb2f3e23ed9dcf6baa905ba585ac7d631613cddc299c4c83463ae29";
+    sha256 = "0adfccscii4rqbfkqq9isv3mln5s0nlvmxnwv4zf5wxj1hkg97hd";
   };
 
-  nativeBuildInputs = [
-    pkgconfig
-  ];
+  doCheck = false;
 
-  buildInputs = [
-    gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM
-    which
-  ];
+  nativeBuildInputs = [ pkgconfig which doxygen
+    ]
+    ++ (lib.optionals (!stdenv.isDarwin) [ wxGTK libX11 ])
+    ++ (lib.optionals stdenv.isDarwin [ wxmac darwin.apple_sdk.frameworks.Cocoa ]);
+
+  buildInputs = with python3Packages; [ gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM ncurses
+      requests libpng gstreamer wxGTK
+      /* webkitgtk gstreamer-plugins-base */
+  ]
+    ++ (lib.optionals (!stdenv.isDarwin) [  (wxGTK.gtk) ])
+    ++ (lib.optional openglSupport pyopengl);
+
+  hardeningDisable = [ "format" ];
+
+  preConfigure = lib.optionalString (!stdenv.isDarwin) ''
+    export DOXYGEN=${doxygen}/bin/doxygen
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace 'cairoLib = None' 'cairoLib = ctypes.CDLL("${cairo}/lib/libcairo.so")'
+    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
+      --replace '_dlls = dict()' '_dlls = {k: ctypes.CDLL(v) for k, v in [
+        ("gdk",        "${wxGTK.gtk}/lib/libgtk-x11-2.0.so"),
+        ("pangocairo", "${pango.out}/lib/libpangocairo-1.0.so"),
+        ("appsvc",     None)
+      ]}'
+  '';
+
+  buildPhase = ''
+  python build.py -v --use_syswx dox etg --nodoc sip build_py
```

Comment:
Are you sure you need to override the whole phase? Can you use `setupPyBuildFlags` instead?

Reply:
huh, cool! I'll try it

Reply:
So I tried `setupPyBuildFlags = ["-v" "--use_syswx" "dox" "etg" "--nodoc" "sip"];` but got 
```
usage: nix_run_setup [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: nix_run_setup --help [cmd1 cmd2 ...]
   or: nix_run_setup --help-commands
   or: nix_run_setup cmd --help

error: option --use_syswx not recognized
```